faculty_cli/cli.py

```diff
@@ -473,55 +505,70 @@ def server():
 
-    if verbose:
-        if not servers:
-            click.echo("No servers.")
-        else:
-            headers = (
-                "Server Name",
-                "Type",
-                "Machine Type",
-                "CPUs",
-                "RAM",
-                "Status",
-                "ID",
-                "Started",
-            )
-            rows = []
-            for server in servers:
-                if isinstance(server.resources, SharedServerResources):
-                    machine_type = "-"
-                    cpus = "{:.3g}".format(server.resources.milli_cpus / 1000)
-                    memory_gb = "{:.3g}GB".format(
-                        server.resources.memory_mb / 1000
-                    )
-                else:
-                    machine_type = server.resources.node_type
-                    cpus = "-"
-                    memory_gb = "-"
-                rows.append(
-                    (
-                        server.name,
-                        server.type,
-                        machine_type,
-                        cpus,
-                        memory_gb,
-                        server.status.value,
-                        server.id,
-                        server.created_at.strftime("%Y-%m-%d %H:%M"),
-                    )
+    status_filter = None if all else ServerStatus.RUNNING
+    headers = (
+        "Project Name",
+        "Project ID",
+        "Server Name",
+        "Type",
+        "Machine Type",
+        "CPUs",
+        "RAM",
+        "Status",
+        "Server ID",
+        "Started",
+    )
+    found_servers = []
+    for project_id, project_name in projects:
+        for server in servers[project_id]:
+            machine_type, cpus, memory_gb = _server_spec(server)
+            found_servers.append(
+                (
+                    project_name,
+                    project_id,
+                    server.name,
+                    server.type,
+                    machine_type,
+                    cpus,
+                    memory_gb,
+                    server.status.value,
+                    server.id,
+                    server.created_at.strftime("%Y-%m-%d %H:%M"),
                 )
-            click.echo(tabulate(rows, headers, tablefmt="plain"))
+            )
+    if not found_servers:
+        print("No servers.")
+    elif provided_project and verbose:
+        servers = [server[2:] for server in found_servers]
+        click.echo(tabulate(servers, headers[2:], tablefmt="plain"))
+    elif provided_project:
+        for server in found_servers:
+            click.echo(server[2])
+    elif verbose:
+        click.echo(tabulate(found_servers, headers, tablefmt="plain"))
     else:
-        for server in servers:
-            click.echo(server.name)
+        servers = [(server[0], server[2]) for server in found_servers]
+        click.echo(
+            tabulate(servers, (headers[0], headers[2]), tablefmt="plain")
```

Comment:
The idea of the non-verbose mode is that there is one server name per line, and so it can be piped to other commands without having to do any parsing. Can we retain that here? If people really need both project and server name, they can combine the existing commands.